presto-main/src/test/java/com/facebook/presto/sql/analyzer/TestMaterializedViewCandidateExtractor.java

```diff
@@ -0,0 +1,219 @@
+
+        Map<QualifiedName, List<QualifiedName>> baseTableToMaterializedViews = new HashMap<>();
+        baseTableToMaterializedViews.put(baseTable1, ImmutableList.of(materializedView1, materializedView2));
+        baseTableToMaterializedViews.put(baseTable2, ImmutableList.of(materializedView3, materializedView4));
+
+        String baseQuerySql = format("SELECT x, y FROM %s JOIN %s ON %s.id = %s.id", baseTable1, baseTable2, baseTable1, baseTable2);
+        Query baseQuery = (Query) sqlParser.createStatement(baseQuerySql);
+
+        MaterializedViewCandidateExtractor materializedViewCandidateExtractor = new MaterializedViewCandidateExtractor(baseTableToMaterializedViews);
+        materializedViewCandidateExtractor.process(baseQuery);
+        Optional<QualifiedName> materializedViewCandidate = materializedViewCandidateExtractor.getMaterializedViewCandidate();
+
+        assertTrue(!materializedViewCandidate.isPresent());
+    }
+
+    @Test
+    public void testMaterializedViewCandidateExtractorWithMultipleJoin()
+    {
+        SqlParser sqlParser = new SqlParser();
+
+        QualifiedName baseTable1 = QualifiedName.of("base_table1");
+        QualifiedName baseTable2 = QualifiedName.of("base_table2");
+        QualifiedName baseTable3 = QualifiedName.of("base_table3");
+        QualifiedName materializedView1 = QualifiedName.of("materialized_view1");
+        QualifiedName materializedView2 = QualifiedName.of("materialized_view2");
+        QualifiedName materializedView3 = QualifiedName.of("materialized_view3");
+
+        Map<QualifiedName, List<QualifiedName>> baseTableToMaterializedViews = new HashMap<>();
+        baseTableToMaterializedViews.put(baseTable1, ImmutableList.of(materializedView1, materializedView2, materializedView3));
+        baseTableToMaterializedViews.put(baseTable2, ImmutableList.of(materializedView1, materializedView3));
+        baseTableToMaterializedViews.put(baseTable3, ImmutableList.of(materializedView2, materializedView3));
+
+        String baseQuerySql = format("SELECT x, y " +
+                "FROM %s b1 " +
+                "JOIN %s b2 ON b1.id = b2.id " +
+                "JOIN %s b3 ON b2.id = b3.id", baseTable1, baseTable2, baseTable3);
+        Query baseQuery = (Query) sqlParser.createStatement(baseQuerySql);
+
+        MaterializedViewCandidateExtractor materializedViewCandidateExtractor = new MaterializedViewCandidateExtractor(baseTableToMaterializedViews);
+        materializedViewCandidateExtractor.process(baseQuery);
+        Optional<QualifiedName> materializedViewCandidate = materializedViewCandidateExtractor.getMaterializedViewCandidate();
+
+        checkState(materializedViewCandidate.isPresent(), "Materialized View candidate is not presented");
+
+        assertEquals(materializedView3, materializedViewCandidate.get());
+    }
+
+    @Test
+    public void testMaterializedViewCandidateExtractorWithMultipleJoinWithNoIntersection()
```

Comment:
this name seems super long, may be we can get rid of tMaterializedViewCandidateExtractor from the method name for all test cases,